daml-script/dump/src/main/scala/com/daml/script/dump/Encode.scala

```diff
@@ -68,13 +69,11 @@ private[dump] object Encode {
       Doc.hardLine +
       Doc.text("dump : Parties -> Script ()") /
       (Doc.text("dump Parties{..} = do") /
-        Doc.stack(
-          sortedAcs.map(createdEvent =>
-            encodeSubmitCreatedEvents(partyMap, cidMap, cidRefs, Seq(createdEvent))
-          ) ++
-            trees.map(t => encodeTree(partyMap, cidMap, cidRefs, t))
-        ) /
-        Doc.text("pure ()")).hang(2)
+        stackNonEmpty(
```

Comment:
Why do we have to filter empty docs now? I’m not really sure which of them can be empty.

Reply:
`Doc.stack` will include empty items in it's arguments as empty lines, with `hang` annoyingly prefixed by whitespace. If we don't generate an ACS, i.e. `--start start`, then `encodeACS` will return an empty doc. So, without the filter the dump would include an empty line at the top. Before, this wasn't an issue because we used `++` on sequences. So, if `sortedAcs` was empty, then the sequence passed to `Doc.stack` would just not include any ACS lines.